docs/architecture/deep-dives/extension-persistence/index.md

```diff
@@ -0,0 +1,130 @@
+well-constrained. The cache of route history and form data is cleared when any of the following
+events occur:
+
+- The account is locked
+- The account is logged out
+- Account switching is used to switch the active account
+- The extension popup has been closed for 2 minutes
+
+In addition, cached form data is cleared when a browser extension navigation event occurs (e.g.
+switching between tabs in the extension).
+
+## Types of persistence
+
+### Route history persistence
+
+Route history is persisted on the extension automatically, with no specific implementation required
+on any component.
+
+The persistence layer ensures that the popup will open at the same route as was active when it
+closed, provided that none of the lifetime expiration events have occurred.
+
+:::tip Excluding a route
+
+If a particular route should be excluded from the history and not persisted, add
+`doNotSaveUrl: true` to the `data` property on the route.
+
+:::
+
+### View data persistence
+
+Route persistence ensures that the user will land back on the route that they were on when the popup
+closed, but it does not persist any state or form data that the user may have modified. In order to
+persist that data, the component is responsible for registering that data with the
+[`ViewCacheService`](https://github.com/bitwarden/clients/blob/main/libs/angular/src/platform/abstractions/view-cache.service.ts).
+This is done prescriptively to ensure that only necessary data is cached and that it is done with
+intention by the component.
+
+The `ViewCacheService` provides an interface for caching both individual state and `FormGroup`s.
+
+#### Caching individual data elements
+
+For individual pieces of state, use the `signal()` method on the `ViewCacheService` to create a
+writeable [signal](https://angular.dev/guide/signals) wrapper around the desired state.
+
+```typescript
+const mySignal = this.viewCacheService.signal({
+    key: "my-state-key"
+    initialValue: null
+});
+```
+
+If a cached value exists, the returned signal will contain the cached data.
+
+Setting the value should be done through the signal's `set()` method:
+
+```typescript
+const mySignal = this.viewCacheService.signal({
+    key: "my-state-key"
+    initialValue: null
+});
+mySignal.set("value")
+```
+
+:::note Equality comparison
+
+By default, signals use `Object.is` to determine equality, and `set()` will only trigger updates if
+the updated value is not equal to the current signal state. See documentation
+[here](https://angular.dev/guide/signals#signal-equality-functions).
+
+:::
+
+Putting this together, the most common implementation pattern would be:
+
+1. **Register the signal** using `ViewCacheService.signal()` on initialization of the component or
+   service responsible for the state being persisted
+2. **Restore state from the signal.** If cached data exists, the signal will contain that data. The
```

Comment:
```suggestion
2. **Restore state from the signal:** If cached data exists, the signal will contain that data. The
```

I like to put colons after each of these, but this is a period which is inconsistent from the others.

I also like to end all bullets with periods, like they are sentences.